jakarta.jdt/org.eclipse.lsp4jakarta.jdt.core/src/main/java/org/eclipse/lsp4jakarta/jdt/core/websocket/WebSocketDiagnosticsCollector.java

```diff
@@ -199,16 +201,97 @@ private void uriMismatchWarningCheck(IType type, List<Diagnostic> diagnostics, I
     }
 
     /**
-     * Create an error diagnostic if a ServerEndpoint annotation does not follow the rules.
+     * Creates an error diagnostic if there exists more than one method annotated
+     * with @OnMessage for a given message format.
+     * 
+     * @param type
+     * @param diagnostics
+     * @param unit
+     * @throws JavaModel
+     */
+    private void onMessageWSMessageFormats(IType type, List<Diagnostic> diagnostics, ICompilationUnit unit)
+            throws JavaModelException {
+        IMethod[] typeMethods = type.getMethods();
+        IAnnotation onMessageTextUsed = null;
+        IAnnotation onMessageBinaryUsed = null;
+        IAnnotation onMessagePongUsed = null;
+        for (IMethod method : typeMethods) {
+            IAnnotation[] allAnnotations = method.getAnnotations();
+            for (IAnnotation annotation : allAnnotations) {
+                if (annotation.getElementName().equals(WebSocketConstants.ON_MESSAGE)) {
+                    ILocalVariable[] allParams = method.getParameters();
+                    for (ILocalVariable param : allParams) {
+                        if (!isParamPath(param)) {
+                            String signature = param.getTypeSignature();
+                            String formatSignature = signature.replace("/", ".");
+                            String resolvedTypeName = JavaModelUtil.getResolvedTypeName(formatSignature, type);
+                            if (resolvedTypeName == null) {
+                                resolvedTypeName = Signature.getSignatureSimpleName(signature);
```

Comment:
`Signature.getSignatureSimpleName` does not return the whole class name. Example for String, `JavaModelUtil.getResolvedTypeName` returns `java.lang.String` whereas `Signature.getSignatureSimpleName` returns `String`

Reply:
alright thanks for letting me know